mergify_engine/debug.py

```diff
@@ -319,8 +320,27 @@ async def report(
     return client
 
 
+def _filter(args, func):
```

Comment:
My head. Ooch. It hurts.

If you want to validate and parse and URL, there are methods in Python stdlib to do that. No need to reinvent the wheel.

Use them!